### DISCUSSION TITLE
Error while using class MessagePassing

### DISCUSSION THREAD

**Discussion by rmngllnn** [pyg-team/pytorch_geometric#4548] [Q&A]
Hello,

I'm trying to run a repo which use PyG. repo is given with data, training checkpoint, precise command to run ... so I'm assuming it's "plug and play". When running the repo, an error occured. As I trace the error, I see it's in the `MessagePassing` class. I see `nod_dim` is set by default to `-2`.

I guess, normally, it's updated with graph construction of the data but it's not. Could you see what could cause this error ?

This is the error I get :

```python
 File "C:\...\Python\Python39\lib\site-packages\torch_geometric\nn\conv\message_passing.py", line 201, in __collect__
    self.__set_size__(size, dim, data)
  File "C:\...\Python\Python39\lib\site-packages\torch_geometric\nn\conv\message_passing.py", line 163, in __set_size__
    size[dim] = src.size(self.node_dim)
IndexError: Dimension out of range (expected to be in range of [-1, 0], but got -2)```

I hope I'm clear and if not I can give more information 😄

**Comment by rusty1s** [comment]
You may want to ensure that your node features inside `data` are two-dimensional: `assert data.x.dim() == 2`.
